tests/test_types.py

```diff
@@ -4127,50 +4202,6 @@ class Model(BaseModel):
     assert Model(v=deque((1, 2, 3))).model_dump_json() == '{"v":[1,2,3]}'
 
 
-def test_deque_any_maxlen():
```

Comment:
I removed support for "preserving" maxlen on a dequeue.
The issue is that we can only do that in some cases.
In particular, we can't do that in JSON mode.
I think we should stick to our promise of "we only guarantee that we give you what is reflected in the type annotation".
If you want the output dequeue to always have a given maxlen then you need to add a validator that enforces that; we can't know you're intended maxlen just based on `deque[int]`.

Reply:
I think this is a regression — if we receive a value that is of type `deque` and it already has a `maxlen` set, why shouldn't we preserve the value? Obviously if we receive a json object (or are otherwise in JSON mode) it makes sense that it can't be preserved, but if we get an object already of type `deque` during validation I don't see why we shouldn't try to preserve the `maxlen`. I would expect most users using `deque` as an annotation are more concerned with the python-side API anyway.

Is it hard to preserve the maxlen for some reason, or are there cases where it would have to be lost that don't involve conversion to JSON? Considering this was an explicitly raised issue that I recently addressed and closed, I would like to preserve the fix unless it's really problematic.

> I think we should stick to our promise of "we only guarantee that we give you what is reflected in the type annotation".

As far as I'm aware, the value of `maxlen` is an attribute of things of type `deque`, not tied to the type hint. I recognize it's not a perfect analogy, but I think not preserving the value of `maxlen` is more akin to not preserving the value of a field of a BaseModel than it is to returning a value of a less specific type (when a field is annotated as a less specific type).

I recognize it's not preserved under a round-trip to JSON but I think if you are using a deque with a maxlen, you are doing things from the python API anyway.

Reply:
I’m happy to re evaluate, it’s not like it can’t be implemented.

Take the case of passing in a list: you end up with a Deque with no maxlen. Maybe you could argue there is no issue if you’re in Python in strict mode, but I fear that users might assume that the output has a maxlen when it doesn’t.

Reply:
I think that's reasonable since that's what happens for e.g. `deque([1, 2, 3])`. Perhaps a better example would be — what happens with private attributes of a BaseModel during validation? I'm assuming they get preserved (though not 100% sure lol)